assets/js/base/components/product-grid/container.js

```diff
@@ -3,29 +3,34 @@
  */
 import { Component } from 'react';
 import PropTypes from 'prop-types';
+import { addQueryArgs, getQueryArg } from '@wordpress/url';
 
 /**
  * Internal dependencies
  */
 import ProductGrid from './index';
+import withWindow from '../../hocs/with-window';
 
 class ProductGridContainer extends Component {
 	state = {
-		currentPage: 1,
-		orderValue: null,
+		currentPage: this.props.currentPage || 1,
+		orderValue: this.props.orderValue,
 	};
 
 	onPageChange = ( newPage ) => {
 		this.setState( {
 			currentPage: newPage,
 		} );
+		this.props.onPageChange( newPage );
 	};
 
 	onOrderChange = ( event ) => {
+		const orderValue = event.target.value;
 		this.setState( {
 			currentPage: 1,
-			orderValue: event.target.value,
+			orderValue,
 		} );
+		this.props.onOrderChange( orderValue );
```

Comment:
Should we have default callbacks for `onOrderChange` and `onPageChange` (defaultProps) in case they aren't present when the component is used directly?  I realize we're not exporting the component directly (we're only exporting the wrapped container), but I wonder if we _should_ export it directly so it can be used without the window wrapper (in which case the defaults would be needed)?

Reply:
> I wonder if we should export it directly so it can be used without the window wrapper (in which case the defaults would be needed)?

We don't have any current use for this component without the HOC and plugins can't use this component either. In the future, any of these two premises might change, but I wonder if it's worth doing that at this point (not a big deal implementing it, but it will be one more thing to maintain).